src/docfx/build/context/Input.cs

```diff
@@ -205,11 +210,57 @@ public FilePath[] ListFilesRecursive(FileOrigin origin, PathString? dependencyNa
             }
         }
 
-        public void AddGeneratedContent(FilePath file, JToken content)
+        public void AddGeneratedContent(FilePath file, JToken content, string? yamlMime)
+        {
+            Debug.Assert(file.Origin == FileOrigin.Generated);
+
+            _generatedContents.TryAdd(file, (yamlMime, content));
+        }
+
+        public string? GetYamlMimeFromGenerated(FilePath file)
```

Comment:
remove? inline with `ReadMimeFromFile`

Reply:
I think we should keep the `?`.

Reply:
I made this method private but kept the `?`.